### ISSUE TITLE
Benchmark runner is broken

### ISSUE THREAD

**Issue by colesbury** [facebookexperimental/free-threading-benchmarking#14]
A bunch of the nightly runs including 05044a4cbfb1e31fa2d341410f59b20404ab982c, 4346c61a5b8b141b93faa0d96b222f9d533b6a8e, and 163fdb5a47cb7367be5a997fc350c17fa9135f02 ended up committing merge conflicts, which breaks the bench_runner parsing.

I'm not exactly sure what the cause was. Maybe a conflict with the weekly runner running at the same time? I'm going to revert the changes and hope for the best.

**Comment by colesbury** [comment]
Things seem to be in an okay state now.

**Comment by mpage** [comment]
> I'm not exactly sure what the cause was. Maybe a conflict with the weekly runner running at the same time?

Yeah - the weekly job ran and committed results for `2a66dd3`. Then the nightly job ran for the same commit. Benchmark results are stored in a file whose base filename is `bm-{date}-{nickname}-{machine}-{fork}-{ref}-{version}-{cpython_hash}`, so the nightly job added to the same file as the weekly job, and the result had merge conflicts. It looks like we attempt to handle this, but it doesn't work:

https://github.com/faster-cpython/bench_runner/blob/d0569e73e0f993f3da17d561fcdf4ada1e78a076/bench_runner/templates/_benchmark.src.yml#L131-L135

I filed https://github.com/faster-cpython/bench_runner/issues/323 upstream.

I think a better solution would be to store results for each run in a unique file (e.g. by including a uuid in the filename).
